tmk_core/protocol/lufa/lufa.c

```diff
@@ -451,6 +462,30 @@ void EVENT_USB_Device_ControlRequest(void) {
                         ReportData = (uint8_t *)&keyboard_report_sent;
                         ReportSize = sizeof(keyboard_report_sent);
                         break;
+#ifdef LAMPARRAY_ENABLE
+                    case LAMPARRAY_INTERFACE:
+                        switch((USB_ControlRequest.wValue & 0xFF)) {
+                            case LAMPARRAY_REPORT_ID_ATTRIBUTES:
+                            {
+                                static lamparray_attributes_report_t ret = { .report_id = LAMPARRAY_REPORT_ID_ATTRIBUTES };
+                                lamparray_get_attributes(&ret.attributes);
+
+                                ReportData = (uint8_t *)&ret;
+                                ReportSize = sizeof(ret);
+                                break;
+                            }
+                            case LAMPARRAY_REPORT_ID_ATTRIBUTES_RESPONSE:
+                            {
+                                static lamparray_attributes_response_report_t ret = { .report_id = LAMPARRAY_REPORT_ID_ATTRIBUTES_RESPONSE };
+                                lamparray_get_attributes_response(&ret.attributes_response);
+
+                                ReportData = (uint8_t *)&ret;
+                                ReportSize = sizeof(ret);
+                                break;
+                            }
+                        }
+                        break;
```

Comment:
In the LUFA implementation the control endpoint transactions are handled at interrupt time too (unless someone sets `NO_INTERRUPT_CONTROL_ENDPOINT = yes`, but that could result in some problems in the presence of user code that blocks the main QMK loop, and there are no real users of that option in QMK; also there is no point changing the LUFA protocol if the ChibiOS implementation cannot be changed to avoid setup callbacks from interrupts).